tio/bufferedreader_test.go

```diff
@@ -54,6 +55,34 @@ func TestBufferedReaderDelimiter(t *testing.T) {
 	data.expect.Nil(msg)
 }
 
+func TestBufferedReaderMultilineDelimiter(t *testing.T) {
+	data := bufferedReaderTestData{
+		expect: ttesting.NewExpect(t),
+		tokens: []string{
+			"1111-12-06 1\n",
+			"2222-12-06T14:58:44.060 [qtp1860944798-11] a08b3652499144f4ac7bbf0bb12e012f ERROR portal2Service.App - \n" +
+			"java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after 30013ms.\n" +
+			"  at com.zaxxer.hikari.pool.HikariPool.createTimeoutException(HikariPool.java:676)\n" +
+			"  at com.zaxxer.hikari.pool.HikariPool.getConnection(HikariPool.java:190)\n" +
+			"  ... 21 common frames omitted\n",
+			"3333-12-06 3\n",
```

Comment:
In this example - how do you know that the last message is complete without the start of the next message?
In a network or large message scenario it might be that the message is cut in the middle and you have to continue reading. If you don't do this you will end up with two message instead of one.

Reply:
I agree. 
I was limiting my self to reading from a local file and didn't see a big picture with network failures and chunked messages.
Thanks for pointing it out.

Reply:
I guess adding a distinguished 'end of message' to my log messages would be a proper solution.

Reply:
I actually like the idea of looking for the start of a message instead of the end. In some cases you cannot change the log messages and the start of a message most of the time is a quite good indicator. However - dealing with dangling messages is a bit tricky, but not impossible.

That's why I put both methods in the change request serii833#1, so that e.g. gollum can allow both types (and use timeouts to catch dangling messages)